frontend/src/views/GShootList.vue

```diff
@@ -311,15 +313,29 @@ export default {
       shootCustomFields,
     } = useProjectShootCustomFields(projectItem)
 
+    const params = useUrlSearchParams('hash-params')
+    const shootSearch = ref(params.q)
+    const debouncedShootSearch = ref(shootSearch.value)
+
+    watch(params, value => {
+      if (value.q !== shootSearch.value) {
+        debouncedShootSearch.value = shootSearch.value = value.q
+      }
+    })
+
+    watch(shootSearch, value => {
+      params.q = value
+    })
+
```

Comment:
* If I change the hash value in the URL the `searchQuery` is not updated correctly due to the `resetSearchQuery` call in the navigation guards.

* To delete the parameter q in the URL we should set the value of `params.q` to `null`. I would not delete the property of the reactive object.
  
I would propose to change the code like this:
```js
    const params = useUrlSearchParams('hash-params')
    const shootSearch = ref(params.q)
    const debouncedShootSearch = ref(shootSearch.value)

    function setShootSearch (value) {
      shootSearch.value = value
      setDebouncedShootSearch()
    }

    function resetShootSearch () {
      debouncedShootSearch.value = shootSearch.value = params.q
    }

    const setDebouncedShootSearch = debounce(() => {
      debouncedShootSearch.value = shootSearch.value
    }, 500)

    watch(() => params.q, value => {
      if (shootSearch.value !== value) {
        setShootSearch(value)
      }
    })

    watch(shootSearch, value => {
      if (!value) {
        params.q = null
      } else if (params.q !== value) {
        params.q = value
      }
    })
```

Reply:
✅